### ISSUE TITLE
Handle Elasticache upgrade with low downtime

### ISSUE THREAD

**Issue by JulesClaussen** [redis/ioredis#1907]
Hey everyone,

We have been using ioredis for a while now, and it works fine except this specific point.
We have an elasticache Redis OSS on AWS, configured in Cluster mode Disabled, but with MultiAZ enabled, and Failover enabled.
We have one primary node, and one replica node.

Client configuration is quite basic, (Typescript, Nest application) the following:
```
Redis({
    host: env.get('REDIS_HOSTNAME'),
    port: env.get('REDIS_PORT'),
    password: env.get('REDIS_PASSWORD'),
    tls: env.get('REDIS_TLS') === 'true' ? {} : undefined,
    ...(!!dbEnv && isNumber(dbEnv) && { db: parseInt(dbEnv) }),
});
```

Whenever we upgrade the Redis (even for minor and release version), we have a 10 minutes unavailability of the Redis. Upgrade takes around 30 minutes all in all, but the Redis is unavailable for around 10 minutes, throwing error such as: `-READONLY You can't write against a read only replica.`

We have tried using the reconnectOnError, but without success:
```
reconnectOnError(err) {
    const targetError = 'READONLY';
    if (err.message.includes(targetError)) {
        return true;
    }
    return false;
},
```

According to documentation, retryStrategy is supposed to reconnect after a minute, so we haven't tried setting it.

Is there a way to handle this, or is this currently not possible?
Also, is there a way to easily test that? Running a failover manually on AWS console does not reproduce the issue for some reasons. Failover in this specific case happens quickly and application is just failing for about a minute or so.

Cheers,
Jules

**Comment by JulesClaussen** [comment]
@danxmoran nope, unfortunately. We opened a ticket to AWS Support at the time, which was quite useless. And we recently tried to upgrade to Valkey, which they mention is way better at handling 0-downtime upgrades, but still no success.
I believe the package does not handle dns change quite well, and we're lacking an HA URL eventually.

**Comment by weikangchia** [comment]
Just trying luck here, anyone managed to find a work around or a way to simulate it? We are facing the same behavior too when we update our AWS elasticache cluster with failover enabled. There is roughly a 10 minutes window where our application get `READONLY You can't write against a read only replica.` error message.

**Comment by JulesClaussen** [comment]
@weikangchia if you find a way to simulate, let us know please 🙏 AWS support mentioned manually running the failover, but this actually "work" and correctly failover without the READONLY issue.

**Comment by weikangchia** [comment]
Will share here if we find a way.

**Comment by soryy708** [comment]
How do we tell if the problem is with `ioredis` or with `AWS Elasticache`?

**Comment by JulesClaussen** [comment]
Hey, did anyone find a workaround for this by any chance?
Cheers
